From 48ae99256e64c65f4af46acd0a5a7b4994836ee7 Mon Sep 17 00:00:00 2001
From: James Morse <james.morse@arm.com>
Date: Mon, 22 Jan 2024 14:58:23 +0800
Subject: [PATCH 671/960] downstream: arm64: mpam: Select ARCH_HAS_CPU_RESCTRL

Enough MPAM support is present to enable ARCH_HAS_CPU_RESCTRL.
Let it rip^Wlink!

Signed-off-by: James Morse <james.morse@arm.com>
Signed-off-by: Zeng Heng <zengheng4@huawei.com>
Signed-off-by: Liu Shixin <liushixin2@huawei.com>
---
 arch/arm64/Kconfig               | 3 ++-
 arch/arm64/include/asm/resctrl.h | 4 ++++
 drivers/platform/mpam/Kconfig    | 4 +++-
 3 files changed, 9 insertions(+), 2 deletions(-)
 create mode 100644 arch/arm64/include/asm/resctrl.h

diff --git a/arch/arm64/Kconfig b/arch/arm64/Kconfig
index 26ae7341f51a..b0cae87f4c1b 100644
--- a/arch/arm64/Kconfig
+++ b/arch/arm64/Kconfig
@@ -2079,7 +2079,8 @@ config ARM64_TLB_RANGE
 config ARM64_MPAM
 	bool "Enable support for MPAM"
 	select ACPI_MPAM if ACPI
-	select ARM_CPU_RESCTRL
+	select ARCH_HAS_CPU_RESCTRL
+	select RESCTRL_FS
 	help
 	  Memory Partitioning and Monitoring is an optional extension
 	  that allows the CPUs to mark load and store transactions with
diff --git a/arch/arm64/include/asm/resctrl.h b/arch/arm64/include/asm/resctrl.h
new file mode 100644
index 000000000000..044ab0947401
--- /dev/null
+++ b/arch/arm64/include/asm/resctrl.h
@@ -0,0 +1,4 @@
+/* SPDX-License-Identifier: GPL-2.0 */
+/* Copyright (C) 2022 Arm Ltd. */
+
+#include <linux/arm_mpam.h>
diff --git a/drivers/platform/mpam/Kconfig b/drivers/platform/mpam/Kconfig
index 13bd86fc5e58..75f5b2454fbe 100644
--- a/drivers/platform/mpam/Kconfig
+++ b/drivers/platform/mpam/Kconfig
@@ -2,5 +2,7 @@
 # CPU resources, not containers or cgroups etc.
 config ARM_CPU_RESCTRL
 	bool
-	depends on ARM64
+	default y
+	depends on ARM64 && ARCH_HAS_CPU_RESCTRL
+	depends on MISC_FILESYSTEMS
 	select RESCTRL_RMID_DEPENDS_ON_CLOSID
-- 
2.43.0

